src/stores/EndpointConfig/Store.ts

```diff
@@ -135,13 +128,19 @@ const getInitialState = (
     setEncryptedEndpointConfig: (encryptedEndpointConfig) => {
         set(
             produce((state: EndpointConfigState) => {
-                const { endpointSchema, customErrors } = get();
-
                 state.encryptedEndpointConfig = isEmpty(encryptedEndpointConfig)
-                    ? createJSONFormDefaults(endpointSchema)
+                    ? createJSONFormDefaults(state.endpointSchema)
                     : encryptedEndpointConfig;
 
-                populateErrors(encryptedEndpointConfig, customErrors, state);
+                // TODO (endpoint config)
+                // I have NO clue why this is passing the encryptedConfig that is becing passed in and
+                //  not what is set to the state. Especially since it could end up with setting defaults to
+                //  the state value
+                populateErrors(
+                    encryptedEndpointConfig,
+                    state.customErrors,
+                    state
+                );
```

Comment:
@kiahna-tucker - do you have _any_ insight into this? I am totally not sure why this is passing the value it is.

Reply:
I cannot recall off the top of my head and would want to look at the commit history for this file.

Reply:
I looked through the history of the file and it has always been this way.